ibis/backends/dask/tests/execution/test_window.py

```diff
@@ -0,0 +1,679 @@
+import io
+from datetime import date
+from operator import methodcaller
+
+import dask.dataframe as dd
+import numpy as np
+import pandas as pd
+import pytest
+from dask.dataframe.utils import tm
+
+import ibis
+import ibis.common.exceptions as com
+import ibis.expr.datatypes as dt
+from ibis.backends.dask import Backend
+from ibis.backends.dask.execution import execute
+from ibis.udf.vectorized import reduction
+
+
+@pytest.fixture(scope='session')
+def sort_kind():
+    return 'mergesort'
+
+
+default = pytest.mark.parametrize('default', [ibis.NA, ibis.literal('a')])
+row_offset = pytest.mark.parametrize('row_offset', list(map(ibis.literal, [-1, 1, 0])))
+range_offset = pytest.mark.parametrize(
+    'range_offset',
+    [
+        ibis.interval(days=1),
+        2 * ibis.interval(days=1),
+        -2 * ibis.interval(days=1),
+    ],
+)
+
+
+@pytest.fixture
+def row_window():
+    return ibis.window(following=0, order_by='plain_int64')
+
+
+@pytest.fixture
+def range_window():
+    return ibis.window(following=0, order_by='plain_datetimes_naive')
+
+
+@default
+@row_offset
+def test_lead(t, df, row_offset, default, row_window):
+    expr = t.dup_strings.lead(row_offset, default=default).over(row_window)
```

Comment:
I agree that Dask test suite should be able to share the same code as the Pandas test suite. But I think this will be a large change, as we should refactor _ALL_ of the Dask test suite.

We can create test base class with the Pandas tests and have the Dask suite inherit from that. 
Just inheriting from a base class won't be enough for the Dask test suite though, as there's some subtle differences between Dask and Pandas. e.g. Dask dataframes are lazily evaluated, so in the test impls, we must call `.compute()`. 

We can deal with all of this, but we'd need to change the way that Pandas test suite is written. I think this would be best addressed in a separate PR where we can refactor _ALL_ of the Dask test suite at once.

@jreback We can create a follow up issue and schedule the work
@cpcloud Let me know if you're okay with this.